src/cctag/ICCTag.hpp

```diff
@@ -41,6 +41,9 @@ class ICCTag
 
 	virtual ~ICCTag() {}
 
+	virtual ICCTag* clone() const = 0;
```

Comment:
Why make it cloneable? In fact, I'd remove clone also from the concrete class (CCTag) and add const to all getters there.